symbolic-base/src/ZkFold/Symbolic/Compiler/ArithmeticCircuit/Internal.hs

```diff
@@ -204,8 +205,8 @@ instance
       in
         zoom #acSystem . modify $ insert (toVar @a p) (p evalConstVar)
 
-    rangeConstraint (SysVar v) upperBound =
-      zoom #acRange . modify $ insert v upperBound
+    rangeConstraint (SysVar v) l =
+      zoom #acRange . modify $ insert l $ S.singleton v
```

Comment:
This would replace the accumulated set with a singleton, but we need to append to it